### PR TITLE
add suport for 'top new' collections

### PR DESCRIPTION
**Pull request by icarcal**
This PR fixes the following issue:
#345

I have updated the following methods:
`.list`

This change add support for the `top new` collections, since the url for top apps and new apps are different right now.
I've added support to the following URLs:
- https://play.google.com/store/apps/new
- https://play.google.com/store/apps/new/category/{CATEGORY}

I've also added some validations, since the URL for categories only works for FAMILY and GAME related ones.

### CODE REVIEW

**Comment by icarcal** [comment]
The failing test has nothing to do with this PR and is ramdonly failing (#209), probably if you run the tests again it will pass.

File: lib/list.js [hunk 1]
```diff
@@ -41,23 +47,27 @@ function validate (opts) {
   if (opts.age && !R.contains(opts.age, R.values(c.age))) {
     throw Error(`Invalid age range ${opts.age}`);
   }
+}
+
+function defineCollectionUrl (opts) {
```
Comment (facundoolano) [inline lib/list.js]: instead of adding this sort of logic, can you update the object in constants to already contain the path that needs to be called?

File: lib/utils/helpers.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const c = require('../constants');
```
Comment (facundoolano) [inline lib/utils/helpers.js]: I'm not a fan of this sort of generic "helper" modules (specially when this has a single function). This function seems to belong together with the objects in the constants module (which could be renamed to reflect that common responsibility.

**Comment by icarcal** [comment]
@facundoolano I removed the helper and put everything into the constants file as you suggested.

I also tried to keep as close as possible to the collection object that the module already had, so it would not be another breaking change.

File: lib/utils/parseCategoryApps.js [hunk 3]
```diff
@@ -18,8 +18,23 @@ function checkIfCategoryIsReallyPaid (selectedCollection, allCategories) {
   return !firstApps[0].free;
 }
 
-function getParsedCluster (categoryObject, collection) {
-  const collections = {
+function getClusterMappings (collection) {
+  if (c.paths[collection].includes(c.clusters.new)) {
+    return {
+      2: {
```
Comment (facundoolano) [inline lib/utils/parseCategoryApps.js]: these objects look like they could be constants in the constants module or in a mappings module, rather than being generated dynamically inside this function.

File: lib/constants.js [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
 
+const { CLUSTER_BASE_URL } = require('./utils/configurations');
+
+const clusters = {
+  new: 'new',
+  top: 'top'
+};
+
+const collectionPaths = {
+  TOP_FREE: { collection: 'topselling_free', path: `${CLUSTER_BASE_URL}/${clusters.top}` },
```
Comment (facundoolano) [inline lib/constants.js]: I find the code in this module very hard to follow: 

1. It's hard to tell what the final urls look like by looking at the code here and the code where these constants are used
2. the fact that you need to reduce to export these and then process on the other side to extrac collection + path looks suspicious.
3. this collection+path fields here also seem more than necessary. If you really need to map each category constant to the cluster type (top vs new), then why not

```js
 TOP_FREE: { collection: 'topselling_free', cluster: 'top' },
...
NEW_FREE: { collection: 'topselling_new_free', cluster: 'new' },
... 
```
?

File: lib/constants.js [hunk 5]
```diff
@@ -1,5 +1,36 @@
...
+  TOP_GROSSING_GAMES: { collection: 'topselling_grossing_games', path: `${CLUSTER_BASE_URL}/${clusters.top}` },
+  NEW_FREE: { collection: 'topselling_new_free', path: `${CLUSTER_BASE_URL}/${clusters.new}` },
+  NEW_PAID: { collection: 'topselling_new_paid', path: `${CLUSTER_BASE_URL}/${clusters.new}` },
+  NEW_FREE_GAMES: { collection: 'topselling_new_free_games', path: `${CLUSTER_BASE_URL}/${clusters.new}` },
+  NEW_PAID_GAMES: { collection: 'topselling_new_paid_games', path: `${CLUSTER_BASE_URL}/${clusters.new}` }
+};
+
+module.exports.clusters = clusters;
+
+module.exports.paths = Object.keys(collectionPaths)
```
Comment (facundoolano) [inline lib/constants.js]: These look suspicious. We should aim to have very obvious and declarative constants, and then if necessary helper functions to process them as needed when they need to be used.

File: lib/list.js [hunk 6]
```diff
@@ -41,30 +45,30 @@ function validate (opts) {
...
+  const { collection, category } = opts;
+  const path = c.paths[collection];
 
-  opts.lang = opts.lang || 'en';
-  opts.country = opts.country || 'us';
-  opts.num = opts.num || 500;
+  return (category)
```
Comment (facundoolano) [inline lib/list.js]: It's very hard to follow what url is being generated here, and how does it relate to the objects in the constants module.

File: lib/utils/parseCategoryApps.js [hunk 7]
```diff
@@ -18,8 +18,23 @@ function checkIfCategoryIsReallyPaid (selectedCollection, allCategories) {
   return !firstApps[0].free;
 }
 
-function getParsedCluster (categoryObject, collection) {
-  const collections = {
+function getClusterMappings (collection) {
```
Comment (facundoolano) [inline lib/utils/parseCategoryApps.js]: I guess I didn't mention when you first introduced this module, but it is solving a rather complex problem and I'm pretty sure it's something that will likely need to update frequently when google rearranges things in the page so we could use a good comment explaining what it does, what clusters are and how this may need to be edited to map what happens in the page.

File: lib/utils/parseCategoryApps.js (hunk 3 above)
Comment (icarcal) [reply lib/utils/parseCategoryApps.js]: I created a new mapping modules for this.

File: lib/constants.js (hunk 5 above)
Comment (icarcal) [reply lib/constants.js]: Definitely agree with you on this.
So I rewrote the constant module to make it cleaner.

File: lib/utils/parseCategoryApps.js (hunk 7 above)
Comment (icarcal) [reply lib/utils/parseCategoryApps.js]: I add some comments and took time to rename some variables to make it more readable.

**Comment by icarcal** [comment]
@facundoolano made some changes to this PR

**Comment by icarcal** [comment]
@facundoolano do you need any other change to this PR?

**Comment by facundoolano** [comment]
Sorry for the delay, I haven't had time to get back to this. I'll try to do it in the next few days.
